src/mjolnir/ingest_transit.cc

```diff
@@ -672,12 +727,19 @@ void ingest_tiles(const boost::property_tree::ptree& pt,
 }
 
 // connect the stop_pairs that span multiple tiles by processing dangling tiles
-void stitch_tiles(const boost::property_tree::ptree& pt,
+void stitch_tiles(const std::string& transit_dir,
                   const std::unordered_set<GraphId>& all_tiles,
                   std::list<GraphId>& tiles,
                   std::mutex& lock) {
   auto grid = TileHierarchy::GetTransitLevel().tiles;
-  auto transit_dir = pt.get<std::string>("mjolnir.transit_dir");
+
+  // for a missing stop_pair member's onestop_id we had to take the stop's gtfs stop_id
+  // but generated nodes actually have their node_type appended on the onestop_id, so here we
+  // need to remove the node_type before matching a candidate tile's nodes with the missing ones
+  const auto stop_id_from_onestop_id = [](const auto& n_oid, NodeType node_type) {
+    const auto type_found = n_oid.find("_" + to_string(node_type));
+    return type_found != n_oid.npos ? n_oid.substr(0, n_oid.length() - type_found) : n_oid;
+  };
```

Comment:
this lambda will accept a tile node's onestop_id and its node_type. if it finds the node_type in the onestop_id, it'll strip it and return the "base" onestop_id, e.g. in: `"Jena_stopid1_transit_multi_platorm"`, out: `"Jena_stopid1"`.

because when looking for missing stop_pair members, we only have the "base" onestop_id. but it's safe to use, since there can't possibly be both inside a single tile: a non-generated `"Jena_stopid1"` **and** a `"Jena_stopid1_transit_multi_platorm"`, which would lead to random results.